### DISCUSSION TITLE
Can you take over Orchard Core Clusters?

### DISCUSSION THREAD

**Discussion by Piedone** [OrchardCMS/OrchardCore#16120] [General]
@jtkech started https://github.com/OrchardCMS/OrchardCore/pull/13633 to address https://github.com/OrchardCMS/OrchardCore/issues/13636. With JT's passing, if we want to have the Clusters feature, somebody else needs to take over.

Who's interested? @ShaneCourtrille @mbarsott I remember you running a lot of tenants, perhaps you'd benefit from this?

**Comment by rikbosch** [comment]
I can't take over, but maybe using Orleans in combination with YARP is an option to share the clustering information and also rebalance the cluster based on resource consumption (https://learn.microsoft.com/en-us/dotnet/orleans/grains/grain-placement#resource-optimized-placement)?

E.g. let orleans decide which tenant is activated on which machine and dynamically update the YARP configuration based on this.

**Reply by Piedone** [reply]
Hmm, yes, that might be an alternative. JT's clusters is static AFAIK, pins a tenant to a server.

FYI regarding Orleans: https://github.com/OrchardCMS/OrchardCore/issues/2875.

**Comment by hishamco** [comment]
I might have a look to what JT did in the past, coz I already have a Yarp module in OCC, but if @rikbosch is interested it will be great

**Reply by rikbosch** [reply]
interested yes, but room in my calendar no.. unfortunately.

**Reply by hishamco** [reply]
I might have a look but not less than a month coz I'm busy with other OC stuff
